tests/src/testcommon.hh

```diff
@@ -187,139 +186,88 @@ template <typename NonLinearOperator>
-    const auto nodeIndex           = localScalarView.index(localScalarView.tree().localIndex(c))[0];
-    stressVector[nodeIndex]        = toDune(computedResult);
-    for (const auto voigtIndex : std::views::iota(0u, static_cast<unsigned int>(gridDim * (gridDim + 1) / 2))) {
-      const auto FEStressComponent       = stressVector[nodeIndex][voigtIndex];
-      const auto expectedStressComponent = expectedStress[c][voigtIndex];
-      const bool isStressCorrect
-          = Dune::FloatCmp::eq<double, Dune::FloatCmp::CmpStyle::absolute>(FEStressComponent, expectedStressComponent);
-      t.check(isStressCorrect) << "Stress component " << voigtIndex << " at node " << c << " is not correctly computed"
-                               << messageIfFailed;
-    }
-  }
-  return t;
-}
-
-template <typename NonLinearOperator, typename FiniteElement>
-[[nodiscard]] auto checkLinearStressOfSimplex(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
-                                              const std::string& messageIfFailed = "") {
+[[nodiscard]] auto checkLinearStress(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
+                                     const std::string& messageIfFailed = "") {
   constexpr int dim = FiniteElement::Traits::mydim;
 
   static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == dim,
-                "The test to check linear stress is only supported with the powerBasis being two-dimensional");
-  assert(fe.localView().element().type() == Dune::GeometryTypes::simplex(dim)
-         && "The test to check linear stress is only supported for simplices");
+                "The test to check linear stress is only supported with the powerBasis having the same power as the "
+                "element dimension");
+  assert(fe.localView().element().type() == Dune::GeometryTypes::cube(dim)
+         or fe.localView().element().type() == Dune::GeometryTypes::simplex(dim)
+                && "The test to check linear stress is only supported for cubes");
   static_assert(
       std::is_same_v<
           std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
           Dune::Functions::LagrangeNode<std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1>>,
       "The test to check linear stress is only supported for a linear Lagrange basis");
 
-  Dune::TestSuite t("Linear Stress check of" + std::to_string(dim) + "D  simplex element");
+  Dune::TestSuite t("Linear Stress check of" + std::to_string(dim) + "D  cube element");
 
-  Eigen::Vector<double, dim*(dim + 1) / 2> expectedStress;
-  auto& displacement = nonLinearOperator.firstParameter();
+  using namespace Ikarus;
+  auto gridView                = fe.localView().globalBasis().gridView();
+  auto element                 = elements(gridView).begin();
+  const auto& referenceElement = Dune::ReferenceElements<double, dim>::general(element->type());
```

Comment:
you can also get the reference Element via   `referenceElement(basis.localView().tree().element())` you can pass this further above

Reply:
Ah i even think this is not neccesary, we could something like that (out of my head):
```cpp
auto element = fe.getGridElement();
const auto& refElement =Dune::referenceElement(element);
```

Reply:
Yes thats right. Basically you just need a grid element from somewhere.